src/game/vmap/VMapFactory.cpp

```diff
@@ -79,26 +85,32 @@ namespace VMAP
         return result;
     }
 
-    //===============================================
-    // just return the instance
-    IVMapManager* VMapFactory::createOrGetVMapManager()
+    /*
+     * Return the instance.
+     */
+    IVMapManager& VMapFactory::GetVMapManager()
```

Comment:
This is more of a philosophical remark, rather than having issue with your code. I wonder if we arent't shooting ourselves in the foot with these dynamic loading procedures and whether we should not just load it on app start in a synchronous manner. This is because if I remember correctly, MMAP code has a similar possible race condition. Just food for thought.

Reply:
You're correct, I think MMap code has similar issues indeed. It's also on my radar for a future change...